lib/two_factor_authentication/hooks/two_factor_authenticatable.rb

```diff
@@ -11,7 +11,9 @@
 
   if user.respond_to?(:need_two_factor_authentication?) && !bypass_by_cookie
     if auth.session(options[:scope])[TwoFactorAuthentication::NEED_AUTHENTICATION] = user.need_two_factor_authentication?(auth.request)
-      user.send_two_factor_authentication_code
+      if !user.totp_enabled?
+        user.send_new_otp
+      end
```

Comment:
Minor style issue flagged by Rubocop: favor the one-line version of this and favor `unless` over `if` for negative conditions:
```ruby
user.send_new_otp unless user.totp_enabled?
```